velox/functions/sparksql/DateTimeFunctions.cpp

```diff
@@ -0,0 +1,198 @@
+      // time in that timezone.
+      if (args[6]->isConstantEncoding()) {
+        auto constantTzID =
+            util::getTimeZoneID(args[6]
+                                    ->asUnchecked<ConstantVector<StringView>>()
+                                    ->valueAt(0)
+                                    .str());
+        rows.applyToSelected([&](vector_size_t row) {
+          auto timestamp = makeTimeStampFromDecodedArgs(
+              row, year, month, day, hour, minute, micros);
+          setTimestampOrNull(row, timestamp, constantTzID, resultFlatVector);
+        });
+      } else {
+        auto timeZone = decodedArgs.at(6);
+        rows.applyToSelected([&](vector_size_t row) {
+          auto timestamp = makeTimeStampFromDecodedArgs(
+              row, year, month, day, hour, minute, micros);
+          auto tzID =
+              util::getTimeZoneID(timeZone->valueAt<StringView>(row).str());
+          setTimestampOrNull(row, timestamp, tzID, resultFlatVector);
+        });
+      }
+    } else {
+      // Otherwise use session timezone.
+      rows.applyToSelected([&](vector_size_t row) {
+        auto timestamp = makeTimeStampFromDecodedArgs(
+            row, year, month, day, hour, minute, micros);
+        setTimestampOrNull(row, timestamp, sessionTzID_, resultFlatVector);
+      });
+    }
+  }
+
+  static std::vector<std::shared_ptr<exec::FunctionSignature>> signatures() {
+    return {
+        exec::FunctionSignatureBuilder()
+            .integerVariable("precision")
+            .integerVariable("scale")
+            .returnType("timestamp")
+            .argumentType("integer")
+            .argumentType("integer")
+            .argumentType("integer")
+            .argumentType("integer")
+            .argumentType("integer")
+            .argumentType("decimal(precision, scale)")
+            .build(),
+        exec::FunctionSignatureBuilder()
+            .integerVariable("precision")
+            .integerVariable("scale")
+            .returnType("timestamp")
+            .argumentType("integer")
+            .argumentType("integer")
+            .argumentType("integer")
+            .argumentType("integer")
+            .argumentType("integer")
+            .argumentType("decimal(precision, scale)")
+            .argumentType("varchar")
+            .build(),
+    };
+  }
+
+ private:
+  int64_t sessionTzID_;
+};
+
+std::shared_ptr<exec::VectorFunction> createMakeTimestampFunction(
+    const std::string& /* name */,
+    const std::vector<exec::VectorFunctionArg>& inputArgs,
+    const core::QueryConfig& config) {
+  const auto sessionTzName = config.sessionTimezone();
+  VELOX_USER_CHECK(
+      !sessionTzName.empty(),
+      "make_timestamp requires session time zone to be set.")
+  const auto sessionTzID = util::getTimeZoneID(sessionTzName);
+
+  VELOX_USER_CHECK(
+      inputArgs[5].type->isShortDecimal(),
+      "Seconds must be short decimal type but got {}",
+      inputArgs[5].type->toString());
```

Comment:
Maybe add type checks for all input args.

Reply:
I wonder why it would be necessary to check all input args. This decimal check is aligned with Spark's https://github.com/apache/spark/blob/e98872fb5d07d570e6d0516b49a5d2e58876d1a6/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/datetimeExpressions.scala#L2601-L2602